### PR TITLE
Fixup 'latest' handling and how things are uploaded to S3

### PR DESCRIPTION
**Pull request by mikedougherty**
TL;DR: This should take care of the open P1 issues: #28, #29, #34.

* `LATEST=1` for #29: Using this environment variable will result in a copy of the content in two separate directories: one named after the `VERSION` and one named 'latest'. Both copies are synced to S3 (version first, then latest).
* `aws s3 sync --delete` for #29: This is just the addition of a flag that was not provided previously. See the AWS CLI docs here for more info: http://docs.aws.amazon.com/cli/latest/reference/s3/sync.html
* `HUGO_BASE_URL=...` for #34: In order to run `make release`, the HUGO_BASE_URL environment variable **must** be set. The make process will exit with a failure if it is not set.

Putting it all together, you can do a new release to docs.docker.com as follows:

```bash
HUGO_BASE_URL="docs.docker.com" LATEST=1 make release
```

You can leave out `LATEST=1` if you need to (for example) update the v1.5 docs.
***

Note that if we are able to move ahead and remove the 'root' copy of the documentation, we will need to put a file at /index.html which redirects to /latest/index.html

***
@moxiegirl

### CODE REVIEW

**Comment by SvenDowideit** [comment]
the code has moved on, we're in the process of moving away from compose, and for 1.10 we look like we're ditching s3, so I'm going to close this
